File: bin/update.ts
#!/usr/bin/env deno run --allow-read
import { isDebian, isFedora } from "./distro.ts";

(async () => {
  if (isDebian()) {
    await Deno.run({ cmd: ["sudo", "apt-get", "update"] }).status();
    await Deno.run({ cmd: ["sudo", "apt-get", "upgrade", "-y"] }).status();
    await Deno.run({ cmd: ["sudo", "apt-get", "upgrade", "-y"] }).status();
    await Deno.run({ cmd: ["sudo", "apt-get", "autoclean"] }).status();
    await Deno.run({ cmd: ["sudo", "apt-get", "autoremove"] }).status();
    Deno.exit(0);
  } else if (isFedora()) {
    const commands = ["update", "upgrade", "autoremove"];
    for (const cmd of commands) {
      await Deno.run({ cmd: ["dnf", "-y", cmd] }).status();
    }
    await Deno.run({ cmd: ["dnf", "-y", "clean", "packages"] }).status();
    Deno.exit(0);
  }
  console.error("fatal: unsupported system");
  Deno.exit(1);
})();

File: bin/update.rb
#!/usr/bin/env ruby -w
# frozen_string_literal: true

require_relative './distro'

if debian?
  p 'Debian'
  p `sudo apt-get update`
  p `sudo apt-get upgrade -y`
  p `sudo apt-get autoclean`
  p `sudo apt-get autoremove`
  exit 0
elsif fedora?
  %w[update upgrade autoremove clean].each do |command|
    p `dnf #{command} -y`
  end
  p `dnf -y clean packages`
elsif darwin?
  p 'Darwin'
  p `brew update`
  p `brew outdated`
  p `brew upgrade`
  p `brew cleanup`
else
  puts 'fatal: unsupported system'
  exit 1
end
